lib/incoming_mail.rb

```diff
@@ -0,0 +1,110 @@
+module IncomingMail
```

Comment:
> anything that would potentially be re-usable by another app could go in `lib`

I would say anything that would be potentially reusable should be a gem :trollface:

Files under `lib/tasks/` are almost always app-specific, so I'm not sure I agree with that distinction. Honestly, if I were building Rails from scratch, I'm not sure that there's much need for both `app/` and `lib/` directories, other than that certain directories beneath them (`app/views/`) have special meaning.

> this parser, for example, feels very particular to this app. Perhaps we could add it to `app/models/`

Hmm, I think of `app/models/` as specific to classes that deal with retrieving/storing data (and their `concerns/`), which seems like a stretch here.